bot/cogs/information.py

```diff
@@ -121,13 +123,23 @@ def __init__(self, bot: Bot):
 
         await ctx.send(embed=embed)
 
-    @with_role(*MODERATION_ROLES)
     @command(name="user", aliases=["user_info", "member", "member_info"])
     async def user_info(self, ctx: Context, user: Member = None, hidden: bool = False):
         """
         Returns info about a user.
         """
 
+        # Do a role check if this is being executed on
+        # someone other than the caller
+        if user and user != ctx.author:
+            if not with_role_check(ctx, *MODERATION_ROLES):
+                raise BadArgument("Only mods can target other users.")
+
+        # Non-moderators may only do this in #bot-commands
+        if not with_role_check(ctx, *MODERATION_ROLES):
+            if not ctx.channel == Channels.bot:
```

Comment:
don't we need to check the channel ID here or compare against a `discord.Object`?

Reply:
yes we do.